RouteDescriber/RouteMetadataDescriber.php

```
@@ -25,24 +30,59 @@ public function describe(Swagger $api, Route $route, \ReflectionMethod $reflecti
 
             $requirements = $route->getRequirements();
             $compiledRoute = $route->compile();
+            $existingParams = $this->getRefParams($api, $operation);
 
             // Don't include host requirements
             foreach ($compiledRoute->getPathVariables() as $pathVariable) {
                 if ('_format' === $pathVariable) {
                     continue;
                 }
 
-                $parameter = $operation->getParameters()->get($pathVariable, 'path');
+                $paramId = $pathVariable.'/path';
+                $parameter = $existingParams[$paramId] ?? $operation->getParameters()->get($pathVariable, 'path');
```

We should just do nothing if there is a already a ref parameter in my opinion, it would be quite strange to update a global parameter due to a local constraint.

---

The reason I did this here is, if they forget to specify it as required, we will have issues in the generated doc.

I agree that updating global ref would be *magic*, but maybe at least we should throw exception here if it is not set as required?

---

Sure that sounds reasonnable

---

Done